src/main/java/org/mockito/internal/MockitoCore.java

```diff
@@ -71,8 +71,9 @@
 public class MockitoCore {
 
     private static final DoNotMockEnforcer DO_NOT_MOCK_ENFORCER = Plugins.getDoNotMockEnforcer();
-    private static final Set<Class<?>> MOCKABLE_CLASSES =
-            Collections.synchronizedSet(new HashSet<>());
+    private static final Map<Boolean, Set<Class<?>>> MOCKABLE_CLASSES_CACHES = Map.of(
```

Comment:
Rather than using a boolean to store both static and non-static mocks, let's use two separate maps for this.

Reply:
Alternative solution: we map from `MockType` to `Set` here per my other comment on introducing a `MockType` enum parameter

Reply:
If we go with a MockType, I'll do that.